packages/nx/src/command-line/dep-graph.ts

```diff
@@ -365,6 +384,62 @@ export async function generateGraph(
   }
 }
 
+function recalculateBaseAndHead(options: {
+  base: string;
+  head: string;
+  uncommitted: boolean;
+  untracked: boolean;
+  all: boolean;
+  files: string[];
+  isBaseOriginallySet: boolean;
+}) {
+  options.base = options.isBaseOriginallySet ? options.base : undefined;
+  if (!options.base && process.env.NX_BASE) {
+    options.base = process.env.NX_BASE;
+  }
+  if (!options.head && process.env.NX_HEAD) {
+    options.head = process.env.NX_HEAD;
+  }
+
+  if (!options.base) {
+    const nxJson = readNxJson();
+    options.base = nxJson.affected?.defaultBase || 'main';
+  }
+
+  if (options.base) {
+    options.base = getMergeBase(options.base, options.head);
+  }
+}
+
+async function recalculateAffectedProjects(options: {
+  all: boolean;
+  exclude: string[];
+  files: string[];
+  uncommitted: boolean;
+  untracked: boolean;
+  base: string;
+  head: string;
+}): Promise<string[]> {
+  const projectGraph = await createProjectGraphAsync({ exitOnError: true });
+  const affectedGraph = options.all
+    ? projectGraph
+    : await filterAffected(
+        projectGraph,
+        calculateFileChanges(
+          parseFiles(options).files,
+          projectGraph.allWorkspaceFiles,
+          options
+        )
+      );
+  if (options.exclude) {
+    const excludedProjects = new Set(options.exclude);
+    return Object.entries(affectedGraph.nodes)
+      .filter(([projectName]) => !excludedProjects.has(projectName))
+      .map(([, project]) => project.name);
+  }
```

Comment:
I don't know if this is sufficient - Compare this to the exclude handling in `projectsToRun` (inside affected.ts)

```
if (nxArgs.exclude) {
    const excludedProjects = new Set(
      findMatchingProjects(nxArgs.exclude, affectedGraph.nodes)
    );

    return Object.entries(affectedGraph.nodes)
      .filter(([projectName]) => !excludedProjects.has(projectName))
      .map(([, project]) => project);
  }
```

Reply:
I've made the change now :)